### DISCUSSION TITLE
클래스 명 질문(Wrapper클래스)

### DISCUSSION THREAD

**Discussion by fkdl0048** [fkdl0048/Mentoring-Project#4] [Q&A] [answered]
안녕하세요..! 

제가 알고 있는 지식이 맞는 지식인지 궁금하기도 하고 많이 헷갈려서 질문드립니다..!

enum과 Enum의 차이점을 모르겠습니다.

공부할때는 enum만 사용하는데 가끔씩 Enum으로 생성할 때가 있습니다. 

그때 마다 궁금했는데 검색해도 자료가 별로 없는 것 같아서 질문드립니다..

*같은 맥락으로 string과 String의 차이점도 없는 것 같은데* 공부할 때 System자료형의 차이뿐이라고 공부했습니다.

Wrapper 클래스의 일종인가요..?

++ Wrapper클래스의 경우 일어나는 박싱 언방식
박싱 언방식의 경우  struct형식을 reference에 할당할 때 일어나는 과정으로 자동으로 되는 경우와 강제로 형변환해야하는 과정이 있다고 알고 있습니다..! 비용도 비싸다고 배웠는데 Wrapper클래스와 관계와 사용이유가 궁금합니다..

**Comment by jongfeel** [comment, marked as answer]
소문자 enum은 enum type을 정의할 때 쓰는 예약어 같은 것이고
대문자 Enum은 enum type으로 된 객체를 사용할 때 쓰는 type 정의라고 생각해 보시면 이해가 쉬울 것 같습니다.
사실상 같은 거라고 보는게 맞고 enum이 Enum을 가리키는 예약어 같은 거라 볼 수도 있습니다.

enum은 정의할 때 쓰는 예약어라 그 이후에 사용 용도는 없지만 
Enum은 class type으로 정의되어 있고 여러 static method도 정의가 되어 있어서 사용이 가능합니다.

string과 String도 마찬가지죠
string은 unicode char를 정의할 때 쓰는 예약어이고
String은 그 예약어를 class로 정의한 Type입니다.

Visual Studio에서 두 단어의 Definition(F12)을 보면 같은 class를 가리키고 있는 걸 확인할 수 있습니다.

그러므로
Wrapper 개념은 아니고 같은 걸 다른 이름으로 만든 겁니다.
